File: RetailStation.Angular/src/app/components/Shared/models/BranchModel.ts
import { CreatorModel } from "./CreatorModel";

export interface BranchModel extends CreatorModel {
    branchId: number;
    nameAR: string;
    nameEN: string;
    phone: string | null;
    address: string | null;
    regionId: number | null;
    isActive: boolean;
    image: string | null;

    // Time fields
    openingTimeFrom: string | null;
    openingTimeTo: string | null;

    workingTimeAR: string | null;
    workingTimeEN: string



    ///old data
    subscriberId: string;
    code: string;
    displayOrder: number | null;
    isAdminBranch: boolean;
    cityId: number | null;
    drawingsCostCenterId: number | null;
    expensesCostCenterId: number | null;
    email: string;
    fax: string;
    notes: string;
}

File: RetailStation.Entities/DTOs/Operation/BranchModel.cs
using Microsoft.AspNetCore.Http;
using RetailStation.Entities.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetailStation.Entities.DTOs.Operation
{
    public class BranchModel : CreatorModel
    {
        public int? BranchId { get; set; }
        public string NameAR { get; set; }
        public string NameEN { get; set; }
        public string Phone { get; set; }
        public string Address { get; set; }
        public int? RegionId { get; set; }
        public bool IsActive { get; set; } = true;
        public IFormFile Image { get; set; }
        public string ImageURL { get; set; }

        public TimeSpan? OpeningTimeFrom { get; set; }
        public TimeSpan? OpeningTimeTo { get; set; }

        public string WorkingTimeAR { get; set; }
        public string WorkingTimeEN { get; set; }
        public int MerchantId { get; set; }
        public int? TotalCount { get; set; }
    }
}
